### ISSUE TITLE
Unable to complete FIDO2 2FA - SEGV/nil pointer dereference

### ISSUE THREAD

**Issue by 4oo4** [quexten/goldwarden#40]
I've completed all the setup steps and am attempting to login for the first time with `goldwarden vault login`. On my Vaultwarden instance I have both FIDO2 and TOTP configured for 2FA. After the password prompt I get a prompt `Fido2 PIN: Enter your token's PIN`, however I'm not able to complete this. I don't have a PIN set on my Yubikey for FIDO2, and I can't get past this prompt. I've tried submitting it blank (since there is no PIN) and with the actual FIDO2 payload but it crashes the same way. Looking at logs it's mentioning a segfault/nil pointer dereference:

```
 $ goldwarden vault login --email user@vault.example.com
panic: interface conversion: interface {} is nil, not messages.IPCMessage

goroutine 1 [running]:
github.com/quexten/goldwarden/client.UnixSocketClient.SendToAgent({0x8e37a0?}, {0x8e37a0, 0xc00020c660})
	/home/runner/work/goldwarden/goldwarden/client/unixsocketclient.go:63 +0x269
github.com/quexten/goldwarden/cmd.glob..func7(0xc8ee20?, {0x927d03?, 0x2?, 0x2?})
	/home/runner/work/goldwarden/goldwarden/cmd/login.go:28 +0xd8
github.com/spf13/cobra.(*Command).execute(0xc8ee20, {0xc0001a2ea0, 0x2, 0x2})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:987 +0xa91
github.com/spf13/cobra.(*Command).ExecuteC(0xc8b4a0)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x425
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/quexten/goldwarden/cmd.Execute({0x0, 0x0, {0x0, 0x0}, 0x0, {0xc0001a8bd0, 0x2c}, 0x0, 0x0, {0x0, ...}, ...})
	/home/runner/work/goldwarden/goldwarden/cmd/root.go:35 +0x1fb
main.main()
	/home/runner/work/goldwarden/goldwarden/main.go:83 +0x978

$ journalctl --user -u goldwarden

Dec 30 19:15:03 debian systemd[3395]: Started goldwarden.service - "Goldwarden daemon".
Dec 30 19:15:03 debian goldwarden[35621]: [INF] [19:15] [Goldwarden > Keyring] >>> Creating new memguard keyring
Dec 30 19:15:03 debian goldwarden[35621]: [INF] [19:15] [Goldwarden > SSH] >>> SSH Agent listening on /home/user/.goldwarden-ssh-agent.sock
Dec 30 19:15:03 debian goldwarden[35621]: [INF] [19:15] [Goldwarden > Agent] >>> Agent listening on /home/user/.goldwarden.sock...
Dec 30 19:15:03 debian goldwarden[35621]: Blocking, press ctrl+c to continue...
Dec 30 19:18:55 debian goldwarden[35621]: [INF] [19:18] [Goldwarden > Pinentry] >>> Asking for pin |Unlock Goldwarden|Enter the vault PIN|
Dec 30 19:18:59 debian goldwarden[35621]: [INF] [19:18] [Goldwarden > Pinentry] >>> Got pin from user
Dec 30 19:19:02 debian goldwarden[35621]: [INF] [19:19] [Goldwarden > Pinentry] >>> Asking for pin |Bitwarden Password|Enter your Bitwarden password|
Dec 30 19:19:13 debian goldwarden[35621]: [INF] [19:19] [Goldwarden > Pinentry] >>> Got pin from user
Dec 30 19:19:13 debian goldwarden[35621]: [INF] [19:19] [Goldwarden > Pinentry] >>> Asking for pin |Fido2 PIN|Enter your token's PIN|
Dec 30 19:19:18 debian goldwarden[35621]: [INF] [19:19] [Goldwarden > Pinentry] >>> Got pin from user
Dec 30 19:19:19 debian goldwarden[35621]: panic: runtime error: invalid memory address or nil pointer dereference
Dec 30 19:19:19 debian goldwarden[35621]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x733830]
Dec 30 19:19:19 debian goldwarden[35621]: goroutine 278 [running]:
Dec 30 19:19:19 debian goldwarden[35621]: github.com/quexten/goldwarden/agent/bitwarden/twofactor.Fido2TwoFactor({0xc0000341e0, 0x2b}, {0xc0001a2120, 0x2, 0x88fd00?}, 0xc0003c0380?)
Dec 30 19:19:19 debian goldwarden[35621]:         /home/runner/work/goldwarden/goldwarden/agent/bitwarden/twofactor/fido2twofactor.go:80 +0x430
Dec 30 19:19:19 debian goldwarden[35621]: github.com/quexten/goldwarden/agent/bitwarden/twofactor.PerformSecondFactor(0xc000292068, 0x25a?)
Dec 30 19:19:19 debian goldwarden[35621]:         /home/runner/work/goldwarden/goldwarden/agent/bitwarden/twofactor/twofactor.go:26 +0x3e8
Dec 30 19:19:19 debian goldwarden[35621]: github.com/quexten/goldwarden/agent/bitwarden.Perform2FA(0x380?, 0xc0001a2080, 0xc000002180, {0x9ec098, 0xc0000280f0})
Dec 30 19:19:19 debian goldwarden[35621]:         /home/runner/work/goldwarden/goldwarden/agent/bitwarden/auth.go:233 +0xd3
Dec 30 19:19:19 debian goldwarden[35621]: github.com/quexten/goldwarden/agent/bitwarden.LoginWithMasterpassword({0x9ec098, 0xc0000280f0}, {0xc000290018, 0x14}, 0xc000002180, 0xe3a10cfb5cf091df?)
Dec 30 19:19:19 debian goldwarden[35621]:         /home/runner/work/goldwarden/goldwarden/agent/bitwarden/auth.go:109 +0xd97
Dec 30 19:19:19 debian goldwarden[35621]: github.com/quexten/goldwarden/agent/actions.handleLogin({0xc83190?, {0xc000414780?, 0x873b?, 0xeab?}}, 0xc000002180, 0xc00007e280, 0x4?)
Dec 30 19:19:19 debian goldwarden[35621]:         /home/runner/work/goldwarden/goldwarden/agent/actions/login.go:38 +0x245
Dec 30 19:19:19 debian goldwarden[35621]: github.com/quexten/goldwarden/agent/actions.ensureIsNotLocked.func1({0xc0002a69e4?, {0xc000414780?, 0xc00015a507?, 0xa?}}, 0xc0001129a7?, 0x4?, 0xc0000ebf20)
Dec 30 19:19:19 debian goldwarden[35621]:         /home/runner/work/goldwarden/goldwarden/agent/actions/actions.go:95 +0x1b7
Dec 30 19:19:19 debian goldwarden[35621]: github.com/quexten/goldwarden/agent.serveAgentSession({0x9ee470, 0xc0000be028}, {0xc0001a53f4?, 0x4?}, 0xc?, 0x4?)
Dec 30 19:19:19 debian goldwarden[35621]:         /home/runner/work/goldwarden/goldwarden/agent/unixsocketagent.go:67 +0x252
Dec 30 19:19:19 debian goldwarden[35621]: created by github.com/quexten/goldwarden/agent.StartUnixAgent.func8
Dec 30 19:19:19 debian goldwarden[35621]:         /home/runner/work/goldwarden/goldwarden/agent/unixsocketagent.go:282 +0x55
Dec 30 19:19:19 debian systemd[3395]: goldwarden.service: Main process exited, code=exited, status=2/INVALIDARGUMENT

```
I'm using Debian 12 Gnome/Wayland if it matters.

Cheers

**Comment by quexten** [comment]
Thanks for the report! The crash seems to stem from insufficient error checking. I'll change this to fail more gracefully soon.

As I recall, Bitwarden uses the preferred UV, which means the pin is optional, but should be used when available. Since Goldwarden always asks for the pin, it fails in your case. Should be fairly simple to fix.

**Comment by quexten** [comment]
Should be fixed in 476ed11fddf92c5293f0c74f2ec9fd4804fb5ccc.
